app/src/main/java/com/example/peterchu/watplanner/home/HomeFragment.java

```diff
@@ -89,6 +91,48 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
         });
 
         weekView = (WeekView) root.findViewById(R.id.weekViewHome);
+        // The week view has infinite scrolling horizontally. We have to provide the events of a
+        // month every time the month changes on the week view.
+        weekView.setMonthChangeListener(new MonthLoader.MonthChangeListener() {
+            @Override
+            public List<? extends WeekViewEvent> onMonthChange(int newYear, int newMonth) {
+                List<WeekViewEvent> events = new ArrayList<>();
+                // Color each section and type a different color and add to view
+                Random rnd = new Random();
+                for (List<CourseComponent> components : mCourseSchedule) {
```

Comment:
Can we modularize this as CourseDetail and HomeFragment are displaying the same calendar logic